sixel.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gdamore/tcell/v2"
+)
+
+const (
+	gSixelBegin = "\033P"
+)
+
+var (
+	gSixelFiller = '\u2800'
+)
+
+type sixel struct {
+	data string
+}
+
+type sixelScreen struct {
+	xprev, yprev int
+	sixel        *sixel
+	altFill      bool
+	lastFile     string // TODO maybe use hash of sixels instead to flip altFill
+}
+
+func (sxs *sixelScreen) clear() {
+	sxs.sixel = nil
+}
+
+func (sxs *sixelScreen) fillerStyle(filePath string) tcell.Style {
+	if sxs.lastFile != filePath {
+		sxs.altFill = !sxs.altFill
+	}
+
+	if sxs.altFill {
+		return tcell.StyleDefault.Bold(true)
+	}
+	return tcell.StyleDefault
+}
+
+func newSixelScreen() (sxs sixelScreen) {
+	return sxs
+}
+
+func (sxs *sixelScreen) showSixels() {
+	if sxs.sixel == nil {
+		return
+	}
+
+	// XXX: workaround for bug where quitting lf might leave the terminal in bold
+	fmt.Print("\033[0m")
+
+	fmt.Print("\0337")                              // Save cursor position
+	fmt.Printf("\033[%d;%dH", sxs.yprev, sxs.xprev) // Move cursor to position
+	fmt.Print(sxs.sixel.data)                       //
+	fmt.Print("\0338")                              // Restore cursor position
+}
+
+func (sxs *sixelScreen) printSixel(win *win, screen tcell.Screen, st tcell.Style, reg *reg) {
```

Comment:
The value of the `st` argument isn't actually being used (it is immediately assigned the result of `sxs.fillerStyle`) and can be removed.

Reply:
BTW this is what causes the CI build to fail: https://github.com/gokcehan/lf/actions/runs/5832803019/job/15836746571?pr=1211#step:6:60